### ISSUE TITLE
System.ArgumentException: Can only specify one unknown dimension  

### ISSUE THREAD

**Issue by mohsen0311** [Unity-Technologies/barracuda-release#62]
i have siamese network and i converted to onxx successfully . but  i have the error Can only specify one unknown dimension  . my out put shape is (-1,-1,-1,-1) . in where two outputs become reshape the shape become -1 . how handle this ?

**Comment by amirebrahimi** [comment]
Hi @mohsen0311,

I was able to reproduce this error with another [siamese network](https://github.com/harveyslash/Facial-Similarity-with-Siamese-Networks-in-Pytorch/blob/master/Siamese-networks-medium.ipynb)
However, would you be willing to share your onnx file?

Thanks.

**Comment by mohsen0311** [comment]
tanks for answering :)
[https://drive.google.com/file/d/1-1o9USnkGNIXWl4YNUR0RvqQ6ZvyESIX/view?usp=sharing](url)

**Comment by mohsen0311** [comment]
[https://drive.google.com/file/d/1-1o9USnkGNIXWl4YNUR0RvqQ6ZvyESIX/view?usp=sharing](url)
Tanks :)

**Comment by mohsen0311** [comment]
this is my pytorch model :
  
        ` class MyModel(nn.Module):
           def __init__(self):
           super(MyModel, self).__init__()

           base_model1=models.resnet18(pretrained=True)
           for param in base_model1.parameters():
            param.requires_grad = True

           image_modules = list(base_model1.children())[:-1]
           self.modelA = nn.Sequential(*image_modules)
        
           self.fc =nn.Sequential(nn.Linear(1024,1024),
                               nn.Dropout(0.2),
                               nn.Linear(1024,1024),
                               nn.Dropout(0.2),
                               nn.Linear(1024,512),
                               nn.Dropout(0.2),
                               nn.Linear(512,512),
                               nn.Dropout(0.2),
                               nn.Linear(512,256),
                               nn.Dropout(0.2),
                               nn.Linear(256,256),
                               nn.Dropout(0.2),
                               nn.Linear(256,128),
                               nn.Linear(128,6)
                               )
        
          def forward(self, image, depth):
              a = self.modelA(image)
              b = self.modelA(depth)
              x = torch.cat((a.view(a.size(0),-1), b.view(b.size(0),-1)),dim=1)
              x = self.fc(x)
              return x
  `

**Comment by amirebrahimi** [comment]
This has been fixed and will show up in a future release. For now, you can make a fix locally by changing the -1f on [this line](https://github.com/Unity-Technologies/barracuda-release/blob/2b6748448a79538d7b02d3a996d0f261c59b9afb/Barracuda/Editor/ONNXModelImporter.cs#L146) and the next to 0.

**Comment by amirebrahimi** [comment]
This is now included in Release 1.1.0
